NorthstarDedicatedTest/ExploitFixes.cpp

```diff
@@ -347,6 +347,20 @@ KHOOK(
 	return oCrashFunc_ParseUTF8(a1, a2, strData);
 }
 
+// GetEntByIndex (called by ScriptGetEntByIndex) doesn't check for the index being out of bounds when it's
+// above the max entity count. This allows it to be used to crash servers.
+typedef void*(__fastcall* GetEntByIndexType)(int idx);
+GetEntByIndexType GetEntByIndex;
+
+static void* GetEntByIndexHook(int idx)
+{
+	if (idx >= 0x4000)
+	{
+		return nullptr;
```

Comment:
maybe a error message here like "GetEntByIndex {} is out of bounds" or something like that

Reply:
Resolved in 50c919ec9037e9a8bbc9fee888f35f66ce3379de